core/src/datayoga_core/blocks/cassandra/write/block.py

```diff
@@ -54,27 +53,45 @@ def init(self, context: Optional[Context] = None):
     async def run(self, data: List[Dict[str, Any]]) -> Tuple[List[Dict[str, Any]], List[Result]]:
         logger.debug(f"Running {self.get_block_name()}")
 
-        data = write_utils.write_records(data, self.opcode_field, self.execute_upsert, self.execute_delete)
+        records_by_opcode_groups = write_utils.group_records_by_opcode(data, self.opcode_field, self.keys)
+
+        for records_by_opcode in records_by_opcode_groups:
+            opcode = records_by_opcode["opcode"]
+            records: List[Dict[str, Any]] = records_by_opcode["records"]
+
+            logger.debug(f"Total {len(records)} record(s) with {opcode} opcode")
+            if opcode == OpCode.UPDATE.value:
+                self.execute_upsert(records)
+            elif opcode == OpCode.DELETE.value:
+                self.execute_delete(records)
+
         return utils.produce_data_and_results(data)
 
+    def get_future(self, stmt: PreparedStatement, record: Dict[str, Any]) -> Any:
```

Comment:
do we need a method for 2 lines if this is only called in one place? may be more maintainable to just inline

Reply:
It's called from two places.